brilirs/src/ir.rs

```diff
@@ -0,0 +1,419 @@
+use std::ops::{Add, Index};
+
+use bril_rs::{ConstOps, EffectOps, Instruction, Literal, ValueOps};
+use fxhash::FxHashMap;
+
+use crate::error::InterpError;
+
+/// A Newtype for function indexing
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct FuncIndex(pub usize);
```

Comment:
Purely for the fun of it, one interesting experiment to run here would be to see what happens if you pick `u32` vs. `u64` for the various indices. Probably not a big impact, but it could be fun to measure just in case.